modules/KalturaSupport/components/dualScreen/dualScreen.js

```diff
@@ -621,7 +637,133 @@
 					at: location[0]+location[1],
 					of: $( this.getPlayer().getInterface() )
 				});
-			}
+			},
+			getKClient: function () {
+				if (!this.kClient) {
+					this.kClient = mw.kApiGetPartnerClient(this.embedPlayer.kwidgetid);
+				}
+				return this.kClient;
+			},
+			/**
+			 * Searches for the first/next cue point after the current player time
+			 * @returns a matching cue point if found, null otherwise
+			 */
+			getCurrentPlayerViewModeCuePoint: function ( ) {
+
+				var currentTime = this.getPlayer().currentTime *1000;
+				var cuePoints = this.getPlayerViewModeCuePoints();
+				var cuePoint;
+				var duration=this.getPlayer().isLive() ? 0 : this.getPlayer().getDuration() * 1000;
+
+				//assume sortedCuePoints array
+				for ( var i = 0; i < cuePoints.length; i++ ) {
+
+					var startTime = cuePoints[i].startTime;
+
+					if ( (startTime > currentTime) ||  //stop once we found a future slide (or out of range slide)
+						(duration>0 && startTime>duration)) {
+						break;
+					}
+
+					cuePoint=cuePoints[i];
+
+				}
+
+				// NOTE: the code below can assist debugging issues with play view mode sync
+				// But it will polute the log console so keep it commented unless required for debugging purposes
+				//if (cuePoint)
+				//{
+				//	var nextCuePoint = i < cuePoints.length ? cuePoints[i] : null;
+				//	var message= 'dualscreen: current cue point (' + cuePoint.id + ') time ' + new Date(cuePoint.startTime) + ' | server time ' + new Date(currentTime);
+                //
+				//	if (nextCuePoint) message += ' | next cue point (' + nextCuePoint.id + ') time ' + new Date(nextCuePoint.startTime);
+                //
+				//	mw.log(message);
+				//}
+				return cuePoint;
+			},
+			/**
+			 * Gets all the cue points of type player view mode
+			 * @returns {Array}
+			 */
+			getPlayerViewModeCuePoints : function()
+			{
+				var _this = this;
+				var cuePoints = [];
+				if ( this.getPlayer().kCuePoints ) {
+					cuePoints = _this.getPlayer().kCuePoints.getCodeCuePointsByTag({tag : 'player-view-mode', sortDesc : false});
```

Comment:
@esakal you are iterating the code cuepoints every clock tick to get code cuepoints by tag, this happens every 250ms and is very expensive. why not do this only when change happens, e.g. when you know for sure that new code cuepoint have arrived via API call.